### ISSUE TITLE
Transit stations, stops

### ISSUE THREAD

**Issue by bhousel** [osmlab/name-suggestion-index#4765]
tl;dr Adding items for transit stations is going to be tricky or maybe impossible.

---
#### Background

A lot of this project's code works on the assumption that we can identify a thing by a `key, value, name` tuple.
* key = amenity
* value = cafe
* name = star-bucks  (people type all kinds of stuff into OSM)

When we say "name", we don't anymore mean the literal `name` tag, but it can be a name-like value like we find in other tags like `brand`, `operator`, `flag:name`, or whatever.  But the gist of it is that we can somehow match 1 key, 1 value, and 1 "name" to determine what a thing _is_.

---
#### The problem

Transit stations unfortunately don't all work like other tags due to OSM's [public_transport tagging](https://wiki.openstreetmap.org/wiki/Key:public_transport) convention.

Historically you could do this:
* key = amenity
* value = bus_station
* network = SEPTA (we'll use this as the "name")

And tagging like that is still common in OSM, but it has been deprecated in favor of the [Public Transport v2](https://wiki.openstreetmap.org/w/index.php?oldid=625726) tag schema.

* public_transport = platform
* bus = yes

That's 2 k's and 2 v's.  

So my problem is - I don't know how to support this, when a feature type can't be inferred from a single key,value pair.

We could support features right now that use legacy tagging, but I know that people aren't always leaving the legacy `amenity=bus_station` tags around, and some kinds of stations (like light rail and subway) require multiple tags anyway.

---
#### Steps forward
It sounds like we will need move away from using OSM key/value pairs to identify things.
I'm just thinking out loud here at what I might need to change:

* Rewrite parts of the `build_index.js` code so that `k/v` doesn't literally mean k/v, but instead it's just a thing that we make up.  For example, there is code to add an `amenity=cafe` tag if you happen to add an item in that file missing that tag.
* Rewrite the matcher to handle not just a single k/v but I guess an array k/v tag pairs.
* Rewrite the id generator, hopefully preserving all the existing generated ids that hashed a single k/v
* Rewrite the parts of nsi.guide?  It could be less "OSM-tag-centric" and just say "Cafe" instead of "amenity/cafe"
* So we might need a file somewhere of all the mappings between OSM Tags and whatever we consider categories to be.
* Maybe we just go all-in with wikidata ?  Q2175765 is a better identifier than `public_transport=station+tram=yes`

**Comment by bhousel** [comment]
Just dropping some thoughts here.. There was apparently a lot of editing happening on with the public transport wiki pages ([see history](https://wiki.openstreetmap.org/w/index.php?title=Public_transport&action=history))

The result of it is that it turns out `bus=yes` maybe isn't required on a bus stop (it used to be for a while).  So I might try to add these in the short term, but keep some of the "steps forward" in mind for future.
